lib/active_merchant/billing/gateways/ct_payment.rb

```diff
@@ -0,0 +1,258 @@
+        add_terminal_number(post, options)
+        add_operator_id(post, options)
+        add_invoice(post, money, options)
+        add_payment(post, payment)
+        add_address(post, payment, options)
+        add_customer_data(post, options)
+
+        payment.is_a?(String) ? commit('preAuthorizationWithToken', post) : commit('preAuthorization', post)
+      end
+
+      def capture(money, authorization, options={})
+        requires!(options, :order_id)
+        post = {}
+        add_invoice(post, money, options)
+        add_money(post, money)
+        add_customer_data(post, options)
+        transaction_number, authorization_number, invoice_number = split_authorization(authorization)
+        post[:OriginalTransactionNumber] = transaction_number
+        post[:OriginalAuthorizationNumber] = authorization_number
+        post[:OriginalInvoiceNumber] = invoice_number
+
+        commit('completion', post)
+      end
+
+      def refund(money, authorization, options={})
+        requires!(options, :order_id)
+        post = {}
+        add_invoice(post, money, options)
+        add_money(post, money)
+        add_customer_data(post, options)
+        transaction_number, _, invoice_number = split_authorization(authorization)
+        post[:OriginalTransactionNumber] = transaction_number
+        post[:OriginalInvoiceNumber] = invoice_number
+
+        commit('refundWithoutCard', post)
+      end
+
+      def credit(money, payment, options={})
+        requires!(options, :order_id)
+        post = {}
+        add_terminal_number(post, options)
+        add_money(post, money)
+        add_operator_id(post, options)
+        add_invoice(post, money, options)
+        add_payment(post, payment)
+        add_address(post, payment, options)
+        add_customer_data(post, options)
+
+        payment.is_a?(String) ? commit('refundWithToken', post) : commit('refund', post)
+      end
+
+      def void(authorization, options={})
+        post = {}
+        post[:InputType] = 'I'
+        post[:LanguageCode] = 'E'
+        transaction_number, _, invoice_number = split_authorization(authorization)
+        post[:OriginalTransactionNumber] = transaction_number
+        post[:OriginalInvoiceNumber] = invoice_number
+        add_operator_id(post, options)
+        add_customer_data(post, options)
+
+        commit('void', post)
+      end
+
+      def verify(credit_card, options={})
+        requires!(options, :order_id)
+        post = {}
+        add_terminal_number(post, options)
+        add_operator_id(post, options)
+        add_invoice(post,0, options)
+        add_payment(post, credit_card)
+        add_customer_data(post, options)
+
+        commit('verifyAccount', post)
+      end
+
+      def store(credit_card, options={})
+        requires!(options, :email)
+        post = {
+          LanguageCode: 'E',
+          Name: credit_card.name.rjust(50, ' '),
+          Email: options[:email].rjust(240, ' ')
+        }
```

Comment:
NBD Nitpick - this kinda breaks the pattern of creating an empty hash and adding the fields to it piecemeal found in the other action methods and most other gateways.

Reply:
I actually suggested @nfarve do this in the first place. In my opinion, this is easier to read (especially with symbol keys) than the other style, and it's objectively less repetitive. The main reason we do the other style is because you do this pattern:

```ruby
post = {}
add_frob(post, options)
add_quux(post, options)
add_bar(post, options)
...
```

which in this case we're (partially) skipping since we can. Given that, I don't think emulating the old style purely for consistency's sake necessarily makes sense here—but I don't feel strongly about this, either.

Reply:
Ah... well, same here TBH, this is fine.